### PR TITLE
varenderer: layer transform should also consider flip

### PR DESCRIPTION
**Pull request by js0701**
new camera tests add h-flip

Jira:https://jira01.devtools.intel.com/browse/OAM-61084
Tests: Camera Orientation of CTS verifier
Signed-off-by: Lin Johnson <johnson.lin@intel.com>

### CODE REVIEW

File: common/compositor/va/varenderer.cpp [hunk 1]
```diff
@@ -292,7 +292,8 @@ bool VARenderer::Draw(const MediaState& state, NativeSurface* surface) {
   param_.filter_flags = filter_flags_;
 // currently rotation is only supported by VA on Android.
 #ifdef DISABLE_CURSOR_PLANE
```
Comment (kalyankondapally) [inline common/compositor/va/varenderer.cpp]: we should probably change this check to be #ifdef VA_MAJOR_VERSION > 1

File: common/compositor/va/varenderer.cpp [hunk 2]
```diff
@@ -487,18 +488,22 @@ bool VARenderer::UpdateCaps() {
...
+  if(transform & kTransform270)
+    return VA_ROTATION_90;
+  if(transform & kTransform180)
+    return VA_ROTATION_180;
+  if(transform & kTransform90)
+    return VA_ROTATION_270;
+  return VA_ROTATION_NONE;
+}
+
+uint32_t VARenderer::HWCReflectToVA(uint32_t transform) {
```
Comment (kalyankondapally) [inline common/compositor/va/varenderer.cpp]: This function should be added if #ifdef VA_MAJOR_VERSION > 1 is true, can u build test it in u env and see if this gets included

File: common/compositor/va/varenderer.cpp (hunk 1 above)
Comment (kalyankondapally) [reply common/compositor/va/varenderer.cpp]: @xuguangxin Can u help here, how do we ensure we get the right libva versions which includes this params

File: common/compositor/va/varenderer.cpp (hunk 2 above)
Comment (js0701) [reply common/compositor/va/varenderer.cpp]: OK

File: common/compositor/va/varenderer.cpp (hunk 1 above)
Comment (xuguangxin) [reply common/compositor/va/varenderer.cpp]: @kalyankondapally , why we need to keep libva 1.0 code here.  We will remove libva from AIA and AIA will get libva 2.0 based solution in following weeks.

Comment (kalyankondapally) [reply common/compositor/va/varenderer.cpp]: But till than we need to have the code compiling right ?

File: common/compositor/va/varenderer.cpp [hunk 3]
```diff
@@ -487,18 +490,25 @@ bool VARenderer::UpdateCaps() {
...
-            break;
-    }
-    return VA_ROTATION_NONE;
+uint32_t VARenderer::HWCRotationToVA(uint32_t transform) {
+  if(transform & kTransform270)
+    return VA_ROTATION_90;
+  if(transform & kTransform180)
+    return VA_ROTATION_180;
+  if(transform & kTransform90)
+    return VA_ROTATION_270;
```
Comment (kalyankondapally) [inline common/compositor/va/varenderer.cpp]: hmm why do  u change this to 270 ? Is this something u forgot ? why change original code ?

**Comment by kalyankondapally** [comment]
merged

**Pull request closed without merging**